doc/source/coding-style/formatting-tools.rst

```diff
@@ -0,0 +1,241 @@
+a change. If any of the hooks defined in `.pre-commit-config.yaml` fails, you
+will need to fix the failing files, stage the new changes and try to commit
+those again.
+
+If you want to manually run ``pre-commit``, you can execute:
+
+.. code-block:: bash
+
+    pre-commit run --all-files --show-diff-on-failure
+
+Previous command will show the current and expected style of the code if any of
+the hooks fail.
+
+
+Using ``pre-commit``
+~~~~~~~~~~~~~~~~~~~~
+
+Tox
+---
+A tool you may consider to use in your project is `tox`_. This tool is an
+automation tool similar to `Make`_ but with the advantage of allowing to test
+your package in a temporary virtual environment. This guarantees reproducible
+builds, as your package is no longer tested in "local" mode but in isolated
+form.
+
+Configuration for `tox`_ is stored in a ``tox.ini`` file. The minimum
+configuration for a PyAnsys ``py<product>-<library>`` project should be:
+
+.. code-block:: ini
+
+    [tox]
+    description = Default tox environments list
+    envlist =
+        style,{py37,py38,py39,py310}{,-coverage},doc
+    skip_missing_interpreters = true
+    isolated_build = true
+    isolated_build_env = build
+    
+    [testenv]
+    description = Checks for project unit tests and coverage (if desired)
+    basepython =
+        py37: python3.7
+        py38: python3.8
+        py39: python3.9
+        py310: python3.10
+        py: python3
+        {style,reformat,doc,build}: python3
+    setenv =
+        PYTHONUNBUFFERED = yes
+        coverage: PYTEST_EXTRA_ARGS = --cov=ansys.product --cov-report=term --cov-report=xml --cov-report=html
+    deps =
+        -r{toxinidir}/requirements/requirements_tests.txt
+    commands =
+        pytest {env:PYTEST_MARKERS:} {env:PYTEST_EXTRA_ARGS:} {posargs:-vv}
+    
+    [testenv:style]
+    description = Checks project code style
+    skip_install = true
+    deps =
+        pre-commit
+    commands =
+        pre-commit install
+        pre-commit run --all-files --show-diff-on-failure
+    
+    [testenv:doc]
+    description = Check if documentation generates properly
+    deps =
+        -r{toxinidir}/requirements/requirements_doc.txt
+    commands =
+        sphinx-build -d "{toxworkdir}/doc_doctree" doc/source "{toxworkdir}/doc_out" --color -vW -bhtml
+
+
+Previous configuration assumes that you have a ``requirements/`` directory that
+contains a ``requirements_tests.txt`` and a ``requirements_doc.txt``. In
+addition, the ``style`` environment will execute pre-commit, which guarantees
+the usage of this tool in your project.
+
+Installing ``tox``
+~~~~~~~~~~~~~~~~~~
+You can install this tool as any other Python one by running:
+
+.. code-block:: bash
+
+    python -m pip install tox
+
+
+Using ``tox``
+~~~~~~~~~~~~~
+
+The core concept behind `tox`_ are ``environments``. These are similar to
+``Makefile`` rules and highly customizable. Previous configuration ships with
+different environments among which you can find:
```

Comment:
```suggestion
Behind `tox`_ are ``environments``, which are similar to
``Makefile`` rules and highly customizable. The previous configuration ships with
different environments, including:
```

Reply:
I wasn't sure why "previous" was present in "The previous configuration ...." This sentence might need to be clarified.